vsc_pdf_template_transformer/lib/utils/spanning_widget_json_converter.dart

```diff
@@ -0,0 +1,26 @@
+import 'package:json_annotation/json_annotation.dart';
+import 'package:vsc_pdf_template_transformer/models/tpl_rich_text.dart';
+import '../utils/spanning_widget.dart' as sw;
+
+class SpanningWidgetJsonConverter
+    extends JsonConverter<sw.SpanningWidgetBuilder?, Map<String, dynamic>?> {
+  const SpanningWidgetJsonConverter();
+
+  @override
+  sw.SpanningWidgetBuilder? fromJson(Map<String, dynamic>? json) {
+    if (json != null) {
+      switch (json['className']) {
+        case 'TplRichText':
```

Comment:
This should specify all classes that mixin `SpanningWidget` - there are several that you have already built. It looks like these classes will also need to implement SpanningWIdgetBuilder.